code_generator/End2End/GeneratedDependencies/Generated.csproj

```diff
@@ -0,0 +1,92 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <ProjectGuid>{C2342BDE-6EE1-4B79-9187-FE76503D6128}</ProjectGuid>
+    <OutputType>Exe</OutputType>
+    <RootNamespace>End2End</RootNamespace>
+    <AssemblyName>End2End</AssemblyName>
+    <TargetFrameworkVersion>v4.6.1</TargetFrameworkVersion>
+    <FileAlignment>512</FileAlignment>
+    <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
+    <NuGetPackageImportStamp>
+    </NuGetPackageImportStamp>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <DebugSymbols>true</DebugSymbols>
+    <DebugType>full</DebugType>
+    <Optimize>false</Optimize>
+    <OutputPath>bin\Debug\</OutputPath>
+    <DefineConstants>DEBUG;TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <DebugType>pdbonly</DebugType>
+    <Optimize>true</Optimize>
+    <OutputPath>bin\Release\</OutputPath>
+    <DefineConstants>TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+  </PropertyGroup>
+  <PropertyGroup>
+    <StartupObject>
+    </StartupObject>
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="Improbable.CSharp">
```

Comment:
These links make me sad, but I don't really see a way around it outside of building an entire test Unity project, which feels like a bit of overkill.

@jared-improbable how did the SDK do this?

Reply:
All of the generated code tests for the UnitySDK lived in the Unity project, so the code generator tests were *just* testing the nuts and bolts of the underlying systems, not the generated code itself.

Reply:
If we wanted to also start testing the logic in the Translations we'd need to make a Unity project too. I feel like we're going to reduce the amount of generated code over time, rather than increase it, so this doesn't seem like a priority to me.

Reply:
If this is rebased on the 2018.2b upgrade, couldn't all of this just be rolled back into Unity?

Reply:
Just discussed with @zeroZshadow. Says the asmdef stuff is still a bit flaky, and running `spatial codegen` once is fine so I don't think there's a need to fold this back into Unity right now.